### PR TITLE
metadata download button

### PR DESCRIPTION
**Pull request by ChristianBeilschmidt**

### CODE REVIEW

File: projects/wave-core-new/src/lib/layers/layer-list/layer-list-element/layer-list-element.component.ts [hunk 1]
```diff
@@ -151,4 +157,42 @@ export class LayerListElementComponent implements OnDestroy, OnChanges {
...
+                }),
+                last(),
+            )
+            .subscribe({
+                next: (event) => {
+                    if (event.type !== HttpEventType.Response || event.body === null) {
+                        return;
+                    }
+
+                    const zipArchive = new File([event.body], filenameFromHttpHeaders(event.headers) ?? 'metadata.json');
```
Comment (michaelmattig) [inline projects/wave-core-new/src/lib/layers/layer-list/layer-list-element/layer-list-element.component.ts:184]: why would you call a zipFile metadata.json?

Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/layers/layer-list/layer-list-element/layer-list-element.component.ts]: [4d7cd45](https://github.com/geo-engine/geoengine-ui/pull/388/commits/4d7cd4539baaa1d751d6d9c168d8f77f91e994c1)

File: projects/wave-core-new/src/lib/layers/layer-list/layer-list-element/layer-list-element.component.ts [hunk 2]
```diff
@@ -151,4 +157,42 @@ export class LayerListElementComponent implements OnDestroy, OnChanges {
...
+                    const url = window.URL.createObjectURL(zipArchive);
+
+                    // trigger download
+                    const anchor = document.createElement('a');
+                    anchor.href = url;
+                    anchor.download = zipArchive.name;
+                    anchor.click();
+                },
+                error: (error) => {
+                    this.notificationService.error(`File upload failed: ${error.message}`);
```
Comment (michaelmattig) [inline projects/wave-core-new/src/lib/layers/layer-list/layer-list-element/layer-list-element.component.ts:194]: why upload?

Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/layers/layer-list/layer-list-element/layer-list-element.component.ts]: [1cf1a61](https://github.com/geo-engine/geoengine-ui/pull/388/commits/1cf1a61fa8039a1f5796e752bf40359ac3934fc5)

**Review by michaelmattig — APPROVED** [review]

**Comment by michaelmattig** [comment]
bors merge

**Pull request merged**